packages/platform-android/src/types.ts

```diff
@@ -0,0 +1,158 @@
+export type InquirerPromptT = any;
+
+export interface Command {
+  name: string;
+  description?: string;
+  func: (
+    argv: Array<string>,
+    ctx: Config,
+    args: Object,
+  ) => Promise<void> | void;
+  options?: Array<{
+    name: string,
+    description?: string,
+    parse?: (val: string) => any,
+    default?:
+      | string
+      | boolean
+      | number
+      | ((ctx: Config) => string | boolean | number),
+  }>;
+  examples?: Array<{
+    desc: string,
+    cmd: string,
+  }>;
+}
+
+interface PlatformConfig<
+  ProjectParams,
+  DependencyParams,
+  ProjectConfig,
+  DependencyConfig,
+> {
+  projectConfig: (
+    projectRoot: string,
+    projectParams: ProjectParams | void,
+  ) => ProjectConfig | void;
+  dependencyConfig: (
+    dependency: string,
+    params: DependencyParams,
+  ) => DependencyConfig | void;
+  linkConfig: () => {
+    isInstalled: (
+      projectConfig: ProjectConfig,
+      packageName: string,
+      dependencyConfig: DependencyConfig,
+    ) => boolean,
+    register: (
+      name: string,
+      dependencyConfig: DependencyConfig,
+      params: Object,
+      projectConfig: ProjectConfig,
+    ) => void,
+    unregister: (
+      name: string,
+      dependencyConfig: DependencyConfig,
+      projectConfig: ProjectConfig,
+      otherDependencies: Array<DependencyConfig>,
+    ) => void,
+    copyAssets: (assets: string[], projectConfig: ProjectConfig) => void,
+    unlinkAssets: (assets: string[], projectConfig: ProjectConfig) => void,
+  };
+}
+
+export interface Config {
+  // Root where the configuration has been resolved from
```

Comment:
how about using TSDoc comments so that it shows up in dialogs?

Reply:
I've added them. It is not much, but it works.